mlflow/tracking/client.py

```diff
@@ -1850,19 +1853,36 @@ def print_run_info(r):
     def log_inputs(
         self,
         run_id: str,
-        datasets: Optional[Sequence[DatasetInput]] = None,
+        datasets: Union[Optional[Sequence[DatasetInput]], Optional[List[pd.DataFrame]]] = None,
     ) -> None:
         """
         Log one or more dataset inputs to a run.
 
         Args:
             run_id: String ID of the run.
-            datasets: List of :py:class:`mlflow.entities.DatasetInput` instances to log.
+            datasets: List of :py:class:`mlflow.entities.DatasetInput` instances to log or List of pd.Dataframe.
 
         Raises:
             mlflow.MlflowException: If any errors occur.
         """
-        self._tracking_client.log_inputs(run_id, datasets)
+        if isinstance(datasets, list) and all(isinstance(d, pd.DataFrame) for d in datasets):
```

Comment:
This isn't the location to put implementation details. The reference `self._tracking_client` that interfaces directly with the tracking store abstraction would be the place to put logic. 
We keep this public-facing API as separate from the logic of manipulation of data as possible (general separation of concerns). These APIs have a single purpose: to log information or retrieve information from the tracking service.